sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -120,11 +121,12 @@ abstract class JdbcDialect extends Serializable {
    * The SQL query that should be used to truncate a table. Dialects can override this method to
    * return a query that is suitable for a particular database. For PostgreSQL, for instance,
    * a different query is used to prevent "TRUNCATE" affecting other tables.
-   * @param table The name of the table.
+   * @param table The table to truncate
+   * @param cascade (OPTIONAL) Whether or not to cascade the truncation. Default: false
    * @return The SQL query to use for truncating a table
    */
   @Since("2.3.0")
-  def getTruncateQuery(table: String): String = {
+  def getTruncateQuery(table: String, cascade: Boolean = false): String = {
```

Comment:
Could you remove the default value here?
It could be safe to prevent accidental inheritance of this wrong implementation.

Reply:
That would force all dialects to implement this method though, which would lead to unnecessary code duplication. Moreover, removing the default value here would change the public API, as it would force others who have written custom dialects to change calls to this method.

Reply:
That is the main reason I request the change. I don't think this implementation is correct on for all dialects. Anyway, it's only my opinion.

Reply:
I think the default for all dialects should be false, regardless of whether the cascade feature is even supported. And for those for which it is supported, it should default to false. It's a feature that you should only use if you explicitly want to cascade imho.